ui/pages/destinations/add.js

```diff
@@ -0,0 +1,181 @@
+import Head from 'next/head'
+import Router from 'next/router'
+import { useState, useEffect } from 'react'
+import useSWR from 'swr'
+
+import Fullscreen from '../../components/modals/fullscreen'
+import HeaderIcon from '../../components/header-icon'
+import InputDropdown from '../../components/input'
+
+const CommandInput = ({ enabledCommandInput, accessKey, currentDestinationName }) => {
+  const server = window.location.host
+  const isHttps = window.location.origin.includes('https')
```

Comment:
Does this work while using self-signed certificates? We may have to use a [trick](https://stackoverflow.com/questions/2618944/check-in-javascript-if-an-ssl-certificate-is-valid) to detect if the certificates are valid (i.e. the user has set up TLS certificates) or not